lib/segment/src/segment_constructor/segment_builder.rs

```diff
@@ -191,34 +229,113 @@ impl SegmentBuilder {
             self.indexed_fields.insert(field, payload_schema);
         }
 
+        id_tracker.mapping_flusher()()?;
+        id_tracker.versions_flusher()()?;
+
         Ok(true)
     }
 
-    pub fn build(
-        mut self,
-        permit: CpuPermit,
-        stopped: &AtomicBool,
-    ) -> Result<Segment, OperationError> {
-        {
+    pub fn build(self, permit: CpuPermit, stopped: &AtomicBool) -> Result<Segment, OperationError> {
+        let (temp_path, destination_path) = {
+            let SegmentBuilder {
+                version,
+                id_tracker,
+                payload_storage,
+                mut vector_storages,
+                segment_config,
+                destination_path,
+                temp_path,
+                indexed_fields,
+            } = self;
+
+            let appendable_flag = segment_config.is_appendable();
+
+            payload_storage.flusher()()?;
+            let payload_storage_arc = Arc::new(AtomicRefCell::new(payload_storage));
+
+            id_tracker.mapping_flusher()()?;
```

Comment:
We don't need to flush `id_tracker` here. it's already flushed in `update_from`

Reply:
Better safe than sorry

Reply:
Yea, thats why I find id_tracker flush in `update_from` save - it's flushed even if `build` is not called. As alternative, I added another comment with proposal to do all flushes in one place just before dropping

Reply:
Id tracker might accumulate changes in RAM if not flushed. I would prefer to keep as is